credentials/credentials.go

```diff
@@ -334,3 +334,12 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 
 	return cfg.Clone()
 }
+
+// RequestInfo contains request data to be attached to the context passed along to GetRequestMetadata calls.
```

Comment:
Please add "`This API is experimental.`" as its own paragraph to all newly-added, exported symbols in this PR.